### PR TITLE
storage: add Bucket/BucketFile "url" attribute

### PR DESCRIPTION
**Pull request by falzm**
Implements #2

### CODE REVIEW

File: exoscale/api/storage.py [hunk 1]
```diff
@@ -343,6 +343,21 @@ def metadata(self):
...
+    @property
+    def url(self):
+        """
+        Stored file URL.
+
+        Returns:
+            str: the stored file URL
+
+        Note:
+            This property value is dynamically retrieved from the API, incurring extra
```
Comment (mcorbin) [inline exoscale/api/storage.py]: Are you sure about that ? It seems to be only a join between the bucket url and the path.

Comment (falzm) [reply exoscale/api/storage.py]: It's not *directly* making an API call but it concatenates the file path to the bucket `zone` property, [which is actually making an API call to get its value](https://github.com/exoscale/python-exoscale/blob/master/exoscale/api/storage.py#L457-L475) so in practice there is an API call being made :)

File: exoscale/api/storage.py [hunk 2]
```diff
@@ -343,6 +343,21 @@ def metadata(self):
...
+            latency.
+        """
+
+        return join(self.bucket.url, self.path)
```
Comment (jgrondier) [inline exoscale/api/storage.py]: Using `os.path.join` for building urls is not a good idea as it uses `os.sep` , you'll get different results depending on your OS. 

You should do something like


```
        p = self.path.replace(os.sep, "/")
        return "/".join([self.bucket.url, p])
```

Comment (falzm) [reply exoscale/api/storage.py]: Good catch, thanks @jgrondier! I've changed for `urllib.parse.urljoin()` instead: 44339bc

Comment (jgrondier) [reply exoscale/api/storage.py]: It still wouldn't work as expected on filesystems with a different path separator.

```
In [10]: url = "https://example.com"                                                                                                                                                

In [11]: windows_path = "var\img"                                                                                                                                                  

In [12]: urljoin(url, windows_path)                                                                                                                                                
Out[12]: 'https://example.com/var\\img'
```

I unfortunately don't have a Windows machine to test interactions, but `pathlib` module might have a solution for that. I don't think I'm seeing something to check that the path is relative either (If it even matters) ?. But I might be mistaken.

Comment (falzm) [reply exoscale/api/storage.py]: Well after reading the `urllib.parse.urljoin()` it doesn't actually does what I want, so finally ended up with a good ol' manual slash stripping: 95b5412

**Pull request merged**